[MIG] l10n_ar_ux: Migration to 16.0

jjscarafia opened the pull request.

jjscarafia committed:
[MIG] l10n_ar_ux: Migration to 16.0

roboadhoc commented:
[Pull request status dashboard](https://runbot.adhoc.com.ar/ingadhoc/odoo-argentina/pull/654).

jjscarafia committed:
[REM] l10n_ar_ux: compatibility with v12- invoice numbers

jjscarafia committed:
[REM] l10n_ar_ux: checkbooks printing funcitonality
Las chequeras fueron deprecadas en el modulo nativo de odoo
Vamos a probar depreciar tmb esta funcionalidad y luego si es necesaria vemos de re-incorporarla.

jjscarafia committed:
[REF] l10n_ar_ux: improve README

jjscarafia committed:
[REF] l10n_ar_ux: disable edition of currency rate till we fix it
No estaria funcionando el cambio al editar la tasa de cambio.
Migramos por el momento sin esta funcionalidad

zaoral commented on l10n_ar_ux/models/account_move.py:
```diff
@@ -139,28 +104,15 @@ def _get_l10n_latam_documents_domain(self):
-            res = super(AccountMove, rec.with_context(force_rate=rec.l10n_ar_currency_rate))._post(soft=soft)
-        res = super(AccountMove, self - other_curr_ar_invoices)._post(soft=soft)
-        return res
+    # def write(self, vals):
```
why do we add this commented code?

jjscarafia replied:
we need to re implement this functionality

zaoral commented on l10n_ar_ux/views/account_move_view.xml:
```diff
@@ -8,21 +8,13 @@
-                <span class="oe_inline o_form_label ml-2 mr-2" groups="base.group_multi_currency" attrs="{'invisible': ['|', ('state','=','draft'), ('other_currency', '=', False)]}">-</span>
-                <span class="oe_inline o_form_label mr-2" groups="base.group_multi_currency" attrs="{'invisible': ['|', ('state','!=','draft'), ('other_currency', '=', False)]}">-</span>
+                <span groups="base.group_multi_currency" attrs="{'invisible': [('other_currency', '=', False)]}">(</span>
                 <field name="l10n_ar_currency_rate" attrs="{'invisible': ['|', '|', ('l10n_ar_currency_rate', '=', False), ('l10n_ar_currency_rate', '=', 0.0), ('other_currency', '=', False)]}"/>
```
we should add country_code to the invisible domain at it is in the line 20.
```suggestion
                <field name="l10n_ar_currency_rate" attrs="{'invisible': [ '|', ('country_code', '!=', 'AR'), '|', '|', ('l10n_ar_currency_rate', '=', False), ('l10n_ar_currency_rate', '=', 0.0), ('other_currency', '=', False)]}"/>
```

jjscarafia replied:
done

zaoral commented on l10n_ar_ux/models/account_move_line.py:
```diff
@@ -1,20 +0,0 @@
-    _inherit = 'account.move.line'
-
-    @api.onchange('currency_id', 'amount_currency')
-    def _onchange_currency(self):
```
In this case, we do not need anymore to ensure that the subtotal are set using the already set l10n_ar_currency_rate instead of the currency actual rate?

jjscarafia replied:
that functionality is deprecated for now and we'll re add later

zaoral commented on l10n_ar_ux/views/account_move_view.xml:
```diff
@@ -8,21 +8,13 @@
-                <span class="oe_inline o_form_label mr-2" groups="base.group_multi_currency" attrs="{'invisible': ['|', ('state','!=','draft'), ('other_currency', '=', False)]}">-</span>
+                <span groups="base.group_multi_currency" attrs="{'invisible': [('other_currency', '=', False)]}">(</span>
                 <field name="l10n_ar_currency_rate" attrs="{'invisible': ['|', '|', ('l10n_ar_currency_rate', '=', False), ('l10n_ar_currency_rate', '=', 0.0), ('other_currency', '=', False)]}"/>
                 <field name="computed_currency_rate" string="Currency Rate (preview)" attrs="{'invisible': ['|', ('l10n_ar_currency_rate', '!=', 0.0), ('other_currency', '=', False)]}"/>
```
add '|', ('country_code', '!=', 'AR'),  to the invisible domain at computed_currency_rate field

jjscarafia replied:
done

zaoral commented on l10n_ar_ux/views/account_move_view.xml:
```diff
@@ -8,21 +8,13 @@
+                <span groups="base.group_multi_currency" attrs="{'invisible': [('other_currency', '=', False)]}">)</span>
+                <!-- TODO arreglar que no se ve el cambio de color. Estos dos botones duplicados son solamente para mostrar cuando la tasa de cambio está forzada. Otra alternativa seria mandarlo en un warning arriba -->
+                <!-- <button name="%(action_account_move_change_rate)d" type="action" icon="fa-pencil" style="color:red" class="btn-link" attrs="{'invisible':['|', '|', '|' , '|', ('state','!=','draft'), ('other_currency', '=', False), ('country_code', '!=', 'AR'), ('l10n_ar_currency_rate', '=', 0.0), ('l10n_ar_currency_rate', '=', False)]}" title="Edit Rate"/>
+                <button name="%(action_account_move_change_rate)d" type="action" icon="fa-pencil" class="btn-link" attrs="{'invisible':['|', '|', '|' , ('state','!=','draft'), ('other_currency', '=', False), ('country_code', '!=', 'AR'), ('l10n_ar_currency_rate', '!=', 0.0)]}" title="Edit Rate"/> -->
```
we are not showing any button to be able to change the rate. we can remove the color part if needed but need to show the button once

jjscarafia replied:
not yet implemeneted

zaoral reviewed (commented):
Hi @jjscarafia 
I made the review and leave you here some comments and suggestions.

Best Regards

jjscarafia commented:
@roboadhoc r- rebase-ff

roboadhoc commented:
Merge method set to rebase and fast-forward.

roboadhoc commented:
I'm sorry, @jjscarafia: r- makes no sense in the current PR state.
